File: llvm-gcc-4.2-2.9/gcc/testsuite/gcc.dg/lvalue-cast-1.c
/* APPLE LOCAL file non lvalue assign */
/* { dg-do compile { target powerpc*-*-darwin* } } */
/* { dg-options "-fnon-lvalue-assign -faltivec" } */
/* { dg-require-effective-target ilp32 } */

int foo(void) {

  char *p;
  long l;
  short s;
  vector unsigned int vui;
  volatile int *pvi;

  (long *)p = &l; /* { dg-warning "target of assignment not really an lvalue" } */
  ((long *)p)++;  /* { dg-warning "target of assignment not really an lvalue" } */
  (short)l = 2;   /* { dg-error "lvalue required as left operand of assignment" } */
  (long)s = 3;    /* { dg-error "lvalue required as left operand of assignment" } */
  (int)pvi = 4;   /* { dg-warning "target of assignment not really an lvalue" } */
  (int)pvi &= 5;  /* { dg-warning "target of assignment not really an lvalue" } */

  (vector float)vui = (vector float)(1.0, 2.0, 3.0, 4.0); /* { dg-warning "target of assignment not really an lvalue" } */

  return 0;
}

File: llvm-gcc-4.2-2.9/gcc/testsuite/objc.dg/lvalue-cast-1.m
/* APPLE LOCAL file non lvalue assign */
/* { dg-do compile } */
/* { dg-options "-fnon-lvalue-assign" } */

#include <objc/Object.h>

typedef struct _NSPoint {
    float x; 
    float y;
} NSPoint;
typedef NSPoint *NSPointPointer;
typedef NSPoint *NSPointArray;
typedef struct _NSSize {
    float width;
    float height;
} NSSize;
typedef struct _NSRect {
     NSPoint origin;
     NSSize size;
} NSRect;

@interface NSLayoutManager: Object {
  NSRect *_cachedRectArray;
}
- (void)_growCachedRectArrayToSize:(unsigned)newSize;
@end

@implementation NSLayoutManager
- (void)_growCachedRectArrayToSize:(unsigned)newSize {
  ( NSRect *)_cachedRectArray = nil;  /* { dg-warning "target of assignment not really an lvalue" } */
}
@end
